moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1873,263 +1874,35 @@ bool RobotState::setFromIKSubgroups(const JointModelGroup* jmg, const EigenSTL::
 
 double RobotState::computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
                                         const LinkModel* link, const Eigen::Vector3d& direction,
-                                        bool global_reference_frame, double distance, const MaxEEFStep& max_step,
-                                        const JumpThreshold& jump_threshold,
-                                        const GroupStateValidityCallbackFn& validCallback,
+                                        bool global_reference_frame, double distance, double max_step,
+                                        double jump_threshold_factor, const GroupStateValidityCallbackFn& validCallback,
                                         const kinematics::KinematicsQueryOptions& options)
 {
-  // this is the Cartesian pose we start from, and have to move in the direction indicated
-  const Eigen::Isometry3d& start_pose = getGlobalLinkTransform(link);
-
-  // the direction can be in the local reference frame (in which case we rotate it)
-  const Eigen::Vector3d rotated_direction = global_reference_frame ? direction : start_pose.rotation() * direction;
-
-  // The target pose is built by applying a translation to the start pose for the desired direction and distance
-  Eigen::Isometry3d target_pose = start_pose;
-  target_pose.translation() += rotated_direction * distance;
-
-  // call computeCartesianPath for the computed target pose in the global reference frame
-  return (distance *
-          computeCartesianPath(group, traj, link, target_pose, true, max_step, jump_threshold, validCallback, options));
+  return CartesianPlanner::computeCartesianPath(this, group, traj, link, direction, global_reference_frame, distance,
```

Comment:
I think that renaming the class to `CartesianInterpolator` effectively removes ambiguity. Personally I hate methods named `solve` or `compute` because it makes it difficult to find function definitions across a code base. I would consider `interpolateCartesianPath` to be a better option but I honestly don't see it as a necessary change.

Reply:
The function definition is easily found via `ClassName::methodName`, isn't it?

Reply:
In this case yes. My comment about methods like `solve` and `compute` is more of a general preference. I still don't see this change as necessary for merge

Reply:
Using a capable IDE, this is a 10s change (refactor -> rename). Please apply it before the merge.